### DISCUSSION TITLE
Named leases

### DISCUSSION THREAD

**Discussion by edhCoder** [etcd-io/etcd#17711] [Ideas]
Has anyone considered having the option to name a lease instead of using the default assigned 16 char hex string.. 

Example: 

etcd lease grant 10 --leasename=my10secondlease
etcd lease grant 10 --leasename=mylease#  (where # is replace with a sequence number)
etcd lease grant 10 --leasename=mylease$  (where $ is replace with a time value)

**Comment by jmhbnz** [comment]
Hey @edhCoder - Thanks for raising this idea, was there more context you could add on why you're interested in this? 🙏🏻

**Reply by edhCoder** [reply]
yes.. just by looking at a lease tag there's no way to know anything about it. A named lease would give users a way to identify what a lease was for and possibly its purpose. Having some variables in the lease name would prevent collisions.  Having an option to list all the leases is meaningless if it's just a bunch of hex strings.
